beacon_node/execution_layer/src/engines.rs

```diff
@@ -238,6 +238,11 @@ impl Engine {
         **self.state.read().await == EngineStateInternal::Synced
     }
 
+    /// Returns `true` if the engine has a status other than synced or syncing.
+    pub async fn is_offline(&self) -> bool {
+        EngineState::from(**self.state.read().await) == EngineState::Offline
```

Comment:
Do you have thoughts on catching `EngineState::AuthFailed` as well? I think the newPayload failures would catch an auth failure in practice. I don't feel strongly either way.

Reply:
I think it's accurate. If the auth is wrong, then the EL is effectively offline.

I imagine this would only be an issue during initial setup, or if resyncing the EE and deleting the JWT secret.